src/js/components/action.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import StripeCheckout from 'react-stripe-checkout';
+
+import { store } from 'stores/app-store';
+import { createTransaction } from 'services/stripe-service';
+import { immediateUpdate } from 'services/user-service';
+
+import { LoadingComponent } from 'components/loading';
+
+export class ActionComponent extends Component {
+    constructor(...args) {
+        super(...args);
+
+        this.state = {
+            state: this.props.state
+        };
+    }
+
+    onStripeToken(token) {
+        let user = store.getState().user;
+        let promises = [];
+        if (!user.email) {
+            promises.push(immediateUpdate({
+                email: token.email
+            }));
+        }
+
+        let transactionPromise = createTransaction(this.props._id, token.id).then(() => {
+            this.setState({
```

Comment:
this part is just local to the component. it will stay there until it's destroyed (I think we never destroy the conversation component, so unless you log out, the component will stay there). 

if the conversation is refetched, this component should retain it's state.. I should probably add something if it's refetched and the new state is `paid` it should take the paid state instead of whatever it is at that moment.